test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -10007,5 +10007,24 @@ TEST_F(AtenXlaTensorTest, TestEmbeddingBackward) {
   }
 }
 
+TEST_F(AtenXlaTensorTest, TestEarlySyncLiveTensors) {
+  torch::Tensor scalar_tensor =
+      torch::scalar_tensor(1., torch::TensorOptions(torch::kFloat));
+  ForEachDevice([&](const torch::Device& device) {
+    torch::Tensor xla_scalar_tensor = CopyToDevice(scalar_tensor, device);
+    torch::Scalar scalar = xla_scalar_tensor.item();
+  });
+  static bool sync =
+      xla::sys_util::GetEnvBool("XLA_SYNC_BEFORE_ITEM_CALL", true);
+  if (sync) {
+    ExpectCounterChanged("EarlySyncLiveTensorsCount",
+                         cpp_test::GetIgnoredCounters());
+
+  } else {
+    ExpectCounterNotChanged("EarlySyncLiveTensorsCount",
+                            cpp_test::GetIgnoredCounters());
+  }
+}
```

Comment:
Maybe also add a
```
ExpectCounterChanged("aten::_local_scalar_dense", cpp_test::GetIgnoredCounters());
```
since we expect `AtenXlaTypeDefault::_local_scalar_dense` to be called in either case.